client/src/app/core/scoped-tokens/scoped-tokens.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, OnInit } from '@angular/core'
+import { HttpClient, HttpHeaders } from '@angular/common/http'
+import { environment } from '../../../environments/environment'
+import { AuthService } from '../auth'
+import { ScopedToken } from '@shared/models/users/user-scoped-token'
+import { Subject } from 'rxjs'
+
+@Injectable()
+export class ScopedTokensService implements OnInit {
+  private static BASE_SCOPED_TOKENS_URL = environment.apiUrl + '/api/v1/users/scoped-tokens'
+
+  private scopedTokens: ScopedToken
+  private scopedTokensSubject = new Subject<ScopedToken>()
+
+  constructor (
+    private http: HttpClient,
+    private authService: AuthService
+  ) {}
+
+  ngOnInit () {
+    this.scopedTokensSubject.asObservable()
+        .subscribe(tokens => {
+          this.scopedTokens = tokens
+        })
+  }
+
+  getScopedTokens (): Promise<ScopedToken> {
+    return new Promise((res, rej) => {
+      if (this.scopedTokens) return res(this.scopedTokens)
+
+      const authHeaderValue = this.authService.getRequestHeaderValue()
+      const headers = new HttpHeaders().set('Authorization', authHeaderValue)
+
+      this.http.get<ScopedToken>(ScopedTokensService.BASE_SCOPED_TOKENS_URL, { headers })
+              .subscribe(
+                scopedTokens => {
+                  this.scopedTokensSubject.next(scopedTokens)
```

Comment:
Just use `this.scopedTokens = scopedTokens` and then remove `scopedTokensSubject`